apollo-ios/Sources/Apollo/GraphQLQueryWatcher.swift

```diff
@@ -5,6 +5,8 @@ import ApolloAPI
 
 /// A `GraphQLQueryWatcher` is responsible for watching the store, and calling the result handler with a new result whenever any of the data the previous result depends on changes.
 ///
+/// NOTE: In order for the `resultHandler` to be called when mutations are executed, you will need to implement Custom Cache Keys - see the Caching section of the Apollo iOS documentation for more details.
```

Comment:
Thanks @damian0815, some suggestions:
* This is not specific, nor limited, to mutations though. If you had another query that used the same fragment you would have the same problem as you experienced before.
* The direction to use custom cache keys is not about having the result handler called but rather about normalizing the response objects that could be shared between operations. The `resultHandler` callback being called is simply the Swift mechanism for getting the result back to your code.

What I'm saying above is that the direction to read about custom cache keys should be about why normalized objects are important for query watchers to function correctly.

Reply:
okay. i am a newbie to apollo, perhaps you could suggest the exact text and i will update the PR accordingly

Reply:
I've pushed two commits with the relevant doc changes.